streamer.c

```diff
@@ -167,32 +168,53 @@ void cLiveStreamer::Close(void)
 void cLiveStreamer::Action(void)
 {
   int ret;
-  sStreamPacket pkt;
-  memset(&pkt, 0, sizeof(sStreamPacket));
-  bool requestStreamChange = false;
+  sStreamPacket pkt_std_data;
+  sStreamPacket pkt_add_data; // Additional data
+  memset(&pkt_std_data, 0, sizeof(sStreamPacket));
+  memset(&pkt_add_data, 0, sizeof(sStreamPacket));
+  bool requestStreamChangeStdData = false;
+  bool requestStreamChangeExtData = false;
   cTimeMs last_info(1000);
   cTimeMs bufferStatsTimer(1000);
 
   while (Running())
   {
-    ret = m_Demuxer.Read(&pkt);
+    ret = m_Demuxer.Read(&pkt_std_data, &pkt_add_data);
     if (ret > 0)
     {
-      if (pkt.pmtChange)
+      if (pkt_std_data.pmtChange)
       {
-        requestStreamChange = true;
+        requestStreamChangeStdData = true;
+        requestStreamChangeExtData = true;
       }
-      if (pkt.data)
+
+      // Process normal data if present
+      if (pkt_std_data.data)
+      {
+        if (pkt_std_data.streamChange || requestStreamChangeStdData)
+          sendStreamChange();
+        requestStreamChangeStdData = false;
+        if (pkt_std_data.reftime)
+        {
+          sendRefTime(&pkt_std_data);
+          pkt_std_data.reftime = 0;
+        }
+        sendStreamPacket(&pkt_std_data);
+      }
+
+      // If some additional data is present inside the stream, it is written there (currently RDS inside MPEG2-Audio)
+      if (pkt_add_data.data)
       {
-        if (pkt.streamChange || requestStreamChange)
+        if (pkt_add_data.streamChange || requestStreamChangeExtData)
```

Comment:
the packet is names add_data, the other parameter ExtData. why not both add or both ext.
(ffmepg has named thos kind of things side_data)